Packs/CohesityHelios/Integrations/CohesityHeliosEventCollector/CohesityHeliosEventCollector.py

```diff
@@ -0,0 +1,343 @@
+        demisto.debug(f'Number of events after de-duping {len(deduped_events)}:{deduped_events}')
+        if not deduped_events:
+            break
+        aggregated_events.extend(deduped_events)
+        # The fetch_end_timestamp variable is used as the next end timestamp and acts like an index marking the earliest alert
+        # pulled so far (alerts are returned in descending order) in cases where we need to perform several calls in the same
+        # round. In cases where we have not reached the user limit we will pass it on to the next fetch-events execution.
+        fetch_end_timestamp = deduped_events[-1].get(time_field_name)
+        ids_for_dedup = get_earliest_event_ids_with_the_same_time(deduped_events, time_field_name)
+        demisto.debug(f'Saved {len(ids_for_dedup)} alert IDs for de-duplication in next iteration')
+        # This means we know there are no more events to pull using the current fetch_start_timestamp and fetch_end_timestamp.
+        if len(temp_events) < PAGE_SIZE:
+            demisto.debug(f'Received {len(temp_events)} events, which is less than {PAGE_SIZE=}')
+            break
+
+    # We only update latest_fetched_event_timestamp if it is empty, o/w it means we are still fetch past events.
+    if not latest_fetched_event_timestamp:
+        latest_fetched_event_timestamp = aggregated_events[0].get(time_field_name) + 1 if aggregated_events else \
+            fetch_end_timestamp
+        demisto.debug(f'latest_fetched_event_timestamp is empty, setting it to \'{latest_fetched_event_timestamp}\'')
+
+    # In case the last events list has less than PAGE_SIZE events we know there are no more events to pull using the current
+    # fetch_start_timestamp and fetch_end_timestamp, so next round is basically a new search. In that case we will:
+    #   1. Update the next_start_timestamp to latest_event_fetched_timestamp
+    #   2. Update ids_for_dedup with the latest_fetched_events_ids
+    #   3. clear the next_end_timestamp, latest_fetched_event_timestamp and latest_fetched_events_ids
+    if len(temp_events) < PAGE_SIZE:
+        next_start_timestamp = latest_fetched_event_timestamp
+        next_end_timestamp = ''
+        latest_fetched_event_timestamp = ''
+        ids_for_dedup = []
+        demisto.debug(f'Last events list has {len(temp_events)} events, which is less than {PAGE_SIZE=}, setting '
+                      f'{next_start_timestamp=}')
+
+    # If we exited the loop and the last list of pulled events is not smaller than PAGE_SIZE (the first if condition)
+    # it must mean that len(events) == PAGE_SIZE, and that we have reached the user limit, implying we are missing more events to
+    # pull from the original window. In that case we will keep the next_start_timestamp as is (fetch_start_timestamp) and will
+    # update the next_end_timestamp to fetch_end_timestamp (the oldest alert fetched). We will also save the list of
+    # ids_for_dedup to be used in the next round
+    else:
```

Comment:
let's add somehow in this case `"nextTrigger": "0"` to the last run (confirmed with Anar we always want to do it in case of in-progress pagination)